app/services/price_policies/time_based_price_calculator.rb

```diff
@@ -40,6 +46,49 @@ def cost_and_subsidy(duration_mins, discount_multiplier)
       end
     end
 
-  end
+    def duration_pricing_cost(duration_mins)
+      duration_in_hours = duration_mins.to_f / 60
+
+      # 3 cases for interval_data:
+      # interval_data = [nil, { min_duration: 5, rate: 8 }]
+      # interval_data = [{ min_duration: 0, rate: 10 }, { min_duration: 5, rate: 8 }]
+      # interval_data = [{ min_duration: 0, rate: 10 }, nil]
```

Comment:
Plan on adding a comment that better explains this if we can't find a solution that's more self explanatory